src/ui/view/dataverse/component/summaryMetadata/summaryMetadata.tsx

```diff
@@ -0,0 +1,108 @@
+import { useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import classNames from 'classnames'
+import { pipe } from 'fp-ts/lib/function'
+import * as A from 'fp-ts/Array'
+import { Link } from 'react-router-dom'
+import { CopyToClipboard } from '@/ui/component/copyToClipboard/copyToClipboard'
+import { Toast } from '@/ui/component/toast/toast'
+import { Icon } from '@/ui/component/icon/icon'
+import type { ItemGeneralMetadata } from '@/ui/view/dataverse/types'
+import './summaryMetadata.scss'
+import './i18n/index'
+
+type SummaryMetadataProps = {
+  metadata: ItemGeneralMetadata[]
+}
+
+type MetadataRowProps = {
+  label: string
+  value: string
+}
+
+const originsProperties = [
+  'publisher',
+  'creator',
+  'registrar',
+  'provider',
+  'belongsTo',
+  'id'
+] as const
+
+type OriginsProperty = (typeof originsProperties)[number]
+
+const isOriginsProperty = (
+  metadata: ItemGeneralMetadata
+): metadata is Omit<ItemGeneralMetadata, 'value'> & { value: string } =>
+  originsProperties.includes(metadata.property as OriginsProperty)
+
+const isAuditMetadata = (
+  metadata: ItemGeneralMetadata
+): metadata is Omit<ItemGeneralMetadata, 'value'> & { value: string } =>
+  metadata.category === 'auditMetadata'
+
+const renderLinkOrSimpleRow = ({ label, value }: MetadataRowProps): JSX.Element =>
+  label === 'belongsTo' ? (
+    <Link className="okp4-dataverse-portal-metadata-link-value" to="/dataverse/dataspace/1">
+      {value}
+    </Link>
+  ) : (
+    <p className="okp4-dataverse-portal-metadata-value">{value}</p>
+  )
+
+const MetadataRow = ({ label, value }: MetadataRowProps): JSX.Element => {
+  const { t } = useTranslation(['metadata', 'common'])
+  const hasClipboard = label === 'registrar' || label === 'createdBy' || label === 'modifiedBy'
+  const [showToast, setShowToast] = useState<boolean>(false)
+  const [hasCopyError, setHasCopyError] = useState<boolean>(true)
+
+  const handleCopy = useCallback((isCopied: boolean): void => {
```

Comment:
I think it's a good idea to make a hook of copyToClipboard but I'm not sure its a good idea to link it to the Toast behavior. Even if we may use this behavior in several places of the app, it may be better to keep the hook less specific. What do you recommend @ccamel @fredericvilcot ?

Reply:
Sure. Incorporating the `useCopyToClipboard()` hook would be good in capturing the behavior and state of the copy function. The signature for this hook could be as follows (fp-ts styled):

```ts
type CopyResult = Either<Error, string>
type CopyFn = (text: string) => Task<CopyResult>

const useCopyToClipboard = (): [CopyResult, CopyFn] => {
  ...
}
```

However I would not add the Toaster there in virtue of the Single Responsibility Principle.

Reply:
IMHO, we must follow [KISS](https://en.wikipedia.org/wiki/KISS_principle) principle.

I see no point in having a hook that manages the copy logic, knowing that for the moment this logic is only used in the CopyToClipboard component and will probably never be used elsewhere...

The only thing that might seem interesting would be to create a high level component that would abstract the use of the toast but in my opinion this is not the immediate priority.

Reply:
@fredericvilcot  brought up a different way of looking at things: 
we could instead consider adding a slice for the `Toast` status in the store. 
This would allow us to position the `Toast` component in our `app.tsx`, retrieve the status of the `Toast` registered in the store and open or close it.
Thanks to this, all the components that use the `Toast` only have to set the status of the `Toast` without having to worry about making it appear in its own JSX.
In the `CopyToClipboard` component, we would keep the logic currently implemented, and in the parent component which uses `CopyToClipboard`, we would set the status of the `Toast` via the slice.
I think we should consider keeping the current implementation of `CopyToClipboard` and `Toast` in order to keep in mind the MVP to be produced, and then in a 2nd time consider the implementation of a new store/slice for the `Toast`, as well as thinking about the possibility of having a `Toast` array (if the user clicks on the copy button and at the same time the result of a query arrives, then the 2 toasts will hide each other, whereas they should be stacked so that the user can see both) 
I'll go with this solution personally because I find it more adapted and optimized. 
What do you think about it @ccamel  & @0xLodz ?

Reply:
Sure.

Reply:
Excellent idea. It makes sense to implement it as a part of the global state. We can do it in a next iteration.